common/trilean/custom_types.go

```diff
@@ -0,0 +1,58 @@
+package trilean
+
+import (
+	"strconv"
+)
+
+type Trilean uint8
+
+const (
+	// This will assign unset to 0, which is the default value in interpolation
+	TriUnset Trilean = iota
+	TriTrue
+	TriFalse
+)
+
+func ToString(t Trilean) string {
+	if t == TriTrue {
+		return "TriTrue"
+	} else if t == TriFalse {
+		return "TriFalse"
+	}
+	return "TriUnset"
+}
+
+func FromString(s string) (Trilean, error) {
+	if s == "" {
+		return TriUnset, nil
+	}
+
+	b, err := strconv.ParseBool(s)
+	if err != nil {
+		return TriUnset, err
+	} else if b == true {
+		return TriTrue, nil
+	} else {
+		return TriFalse, nil
+	}
+}
+
+func FromBool(b bool) Trilean {
+	if b {
+		return TriTrue
+	}
+	return TriFalse
+}
+
+func boolPointer(b bool) *bool {
+	return &b
+}
+
+func ToBoolPointer(t Trilean) *bool {
```

Comment:
Here I think it is better to make it a member:
```suggestion
func (t Trilean) ToBoolPointer() *bool {
```

This reduces slightly the lenght of the line

Reply:
Good point.